rpc/QueryFuture.java

```diff
@@ -21,98 +21,59 @@
 
 import grakn.client.common.exception.GraknClientException;
 import grakn.protocol.TransactionProto;
+import io.grpc.stub.StreamObserver;
 
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import java.util.function.Function;
-import java.util.stream.StreamSupport;
 
-public abstract class QueryFuture<T> implements Future<T> {
-    @Override
-    public abstract T get();
-
-    @Override
-    public abstract T get(long timeout, TimeUnit unit);
-
-    static class Execute<T> extends QueryFuture<T> {
-        private final RPCResponseAccumulator collector;
-        private final Function<TransactionProto.Transaction.Res, T> responseReader;
+public class QueryFuture<T> implements Future<T> {
 
-        public Execute(final RPCResponseAccumulator collector, final Function<TransactionProto.Transaction.Res, T> responseReader) {
```

Comment:
Now that `QueryFuture.Stream` is deleted, `Execute` becomes the only concrete subclass of `QueryFuture`, so we simply dissolve `Execute` and just use `QueryFuture` itself.